core/storage/suggestion/gae_models.py

```diff
@@ -479,17 +480,29 @@ def get_in_review_translation_suggestions(
                 As a user cannot review their own suggestions, suggestions
                 authored by the user will be excluded.
             language_codes: list(str). The list of language codes.
+            exp_ids: list(str). Exploration IDs matching the target ID of the
+                translation suggestions.
 
         Returns:
             list(SuggestionModel). A list of suggestions that are of the given
             type, which are in review, but not created by the given user.
         """
-        return cls.get_all().filter(datastore_services.all_of(
-            cls.status == STATUS_IN_REVIEW,
-            cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
-            cls.author_id != user_id,
-            cls.language_code.IN(language_codes)
-        )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
+
+        if len(exp_ids) > 0:
+            return cls.get_all().filter(datastore_services.all_of(
+                cls.status == STATUS_IN_REVIEW,
+                cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                cls.author_id != user_id,
+                cls.language_code.IN(language_codes),
+                cls.target_id.IN(exp_ids)
+            )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
+        else:
+            return cls.get_all().filter(datastore_services.all_of(
+                cls.status == STATUS_IN_REVIEW,
+                cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                cls.author_id != user_id,
+                cls.language_code.IN(language_codes),
+            )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
```

Comment:
Are you able to create the base query and only add the `cls.target_id.IN(exp_ids)` filter if `len(exp_ids) > 0:`? E.g.: https://cloud.google.com/appengine/docs/standard/python/ndb/queries#filter_by_prop.

Reply:
I have created two seperate functions now, so I believe we won't need to create the base query anymore. What do you think?